proc_CPMG.py

```diff
@@ -3,29 +3,75 @@
 from proc_scripts import postproc_dict
 from proc_scripts.fitting import decay
+import sympy as sp
 fl = fl_mod()
 logger = init_logging('info')
-for searchstr, exp_type, nodename, postproc, label_str, f_range in [
-        #('200221_CPMG_TEMPOLgel_2p9_1','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200304_CPMG_2p6_1','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_3p5_2','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_3p6_2','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        ('200305_CPMG_3p7_2','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_3p7_3','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_3p8_2','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_3p9_2','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
-        #('200305_CPMG_4p0_1','test_equip','signal','spincore_CPMG_v1','deadtime=5',(-500,500)),
+for searchstr, exp_type, nodename, postproc, label_str, f_range, spincore in [
+        #('w8_200917','test_equip',6,'ag_CPMG_strob','water loading 8',(-500,500),False),
+        #('200221_CPMG_TEMPOLgel_2p9_1','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('freeSL_201007','test_equip',3,'ag_CPMG_strob','free SL',(-200,200),False),
+        ('200305_CPMG_3p5_2','test_equip','signal','spincore_CPMG_v1',
+            'deadtime=5',(-500,500),True),
+        #('200305_CPMG_3p6_2','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('200305_CPMG_3p7_2','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('200305_CPMG_3p7_3','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('200305_CPMG_3p8_2','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('200305_CPMG_3p9_2','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
+        #('200305_CPMG_4p0_1','test_equip','signal','spincore_CPMG_v1',
+        #    'deadtime=5',(-500,500),True),
         ]:
-    s =  find_file(searchstr, exp_type=exp_type,
-            expno=nodename, postproc=postproc, lookup=postproc_dict)
+    s = find_file(searchstr, exp_type=exp_type,
+            expno=nodename, postproc=postproc, lookup=postproc_dict, fl=fl)
+    s.ift('t2')
+#{{{Spincore data requires some reordering of dimensions while bruker data does not.
+    #For this reason we make spincore an argument above for the proper processing
+    if spincore:
+        s = s['ph1',1] #the spincore version only has one phasing dimension 'nPhaseSteps'
+        s.mean('nScans')
+        s.reorder('t2',first=True)
+    else:
+        s = s['ph2',-2]['ph1',1]
```

Comment:
still a problem with the spincore here -- as noted in other comments, just inspect the phase to see what phase cycling dimensions are present.